gigascience/gigadb-website
# Homepage news items section rework

# Pull request for issue: #208

NOTE: I suggest to review this PR and https://github.com/gigascience/gigadb-website/pull/2167 simultaneously since #2167 includes refactoring of the carousel added in this PR, mainly the carousel is abstracted to a reusable partial

This is a pull request for the following functionalities:

* full rework of news section in home page

![image](https://github.com/user-attachments/assets/2c99becb-a95d-41c6-83cc-43aa59be7b90)

## How to test?

Visit the home page and verify the following:

- I expect there are no news items to display, so the section should not be displayed at all
- go to protected/components/NewsAndFeedsService.php and in the getTodaysNews function remove the second conditional so that all news are displayed
- refresh the home page. you should see a news section with a carousel and news boxes
- clicking on a "read more" link should open the detail page

alternatively you can create more news items from the admin page http://gigadb.gigasciencejournal.com/news/admin and they should show up in the home page

## How have functionalities been implemented?

- the carousel uses the bootstrap carousel and overrides the styles to match the gigadb colors and to make it look better
- the logic was tweaked to cover a handful of use cases:
  - the carousel is responsive and reactive to the screen width
  - mobile shows one item
  - medium screen shows two items
  - desktop shows three items
  - if the number of items to display is less than the maximum, for example, for desktop, if there are just two news items, rather than showing a blank space, the two news items fill all the space
  - if there are less news items than the maximum that can be displayed at any given time (e.g. 3 for desktop) then the carousel controls are hidden

## Any issues with implementation?

--

## Any changes to automated tests?

tests pass. As far as I have seen, there are no acceptance tests covering this case
